tools/wake-unit/unit.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "unit.h"
+#include "util/colour.h"
+#include "util/diagnostic.h"
+
+#include <set>
+#include <cstring>
+
+DiagnosticReporter *reporter;
+
+struct Test {
+  const char* test_name;
+  TestFunc test;
+  Test(const char* test_name_, TestFunc test_) : test_name(test_name_), test(test_) {}
+};
+
+// TODO: sequester this off to its own file and make it static
+// so that only TestRegister and the test runner can access it.
+std::vector<Test> tests__;
```

Comment:
Why not make this `static` and remove the trailing `__`?
You don't need it to necessarily have its own file; it could still live here while being static.

Reply:
So other files like wake-unit.cpp need to be able to add tests, it just means I need to add a function here to do that so I can make this static instead. I just forgot to address this TODO. This all used to be in wake-unit.cpp and unit.cpp *is* the aforementioned "other" file :)